contracts/nft/TributeERC721.sol

```diff
@@ -0,0 +1,218 @@
+        return _baseURI();
+    }
+
+    function totalSupply() external view returns (uint256) {
+        return _tokenIdCounter.current() - 1;
+    }
+
+    /**
+     * @notice Hashes the provided coupon as an ERC712 hash.
+     * @param dao is the DAO instance to be configured
+     */
+    function hashCouponMessage(
+        DaoRegistry dao,
+        address owner,
+        uint256 nonce
+    ) public view returns (bytes32) {
+        bytes32 message = keccak256(
+            abi.encode(MINT_COUPON_MESSAGE_TYPEHASH, owner, nonce)
+        );
+
+        return hashMessage(dao, address(this), message);
+    }
+
+    function getPriorAmount(address account, uint256 blockNumber)
+        external
+        view
+        returns (uint256)
+    {
+        require(
+            blockNumber < block.number,
+            "nft::getPriorAmount: not yet determined"
+        );
+
+        uint32 nCheckpoints = numCheckpoints[account];
+        if (nCheckpoints == 0) {
+            return 0;
+        }
+
+        // First check most recent balance
+        if (checkpoints[account][nCheckpoints - 1].fromBlock <= blockNumber) {
+            return checkpoints[account][nCheckpoints - 1].amount;
+        }
+
+        // Next check implicit zero balance
+        if (checkpoints[account][0].fromBlock > blockNumber) {
+            return 0;
+        }
+
+        uint32 lower = 0;
+        uint32 upper = nCheckpoints - 1;
+        while (upper > lower) {
+            uint32 center = upper - (upper - lower) / 2; // ceil, avoiding overflow
+            Checkpoint memory cp = checkpoints[account][center];
+            if (cp.fromBlock == blockNumber) {
+                return cp.amount;
+            } else if (cp.fromBlock < blockNumber) {
+                lower = center;
+            } else {
+                upper = center - 1;
+            }
+        }
+        return checkpoints[account][lower].amount;
+    }
+
+    function _createNewAmountCheckpoint(address member) internal {
+        uint256 amount = balanceOf(member);
+
+        uint160 newAmount = uint160(amount);
+
+        uint32 nCheckpoints = numCheckpoints[member];
+        if (
+            // The only condition that we should allow the amount update
+            // is when the block.number exactly matches the fromBlock value.
+            // Anything different from that should generate a new checkpoint.
+            //slither-disable-next-line incorrect-equality
+            nCheckpoints > 0 &&
+            checkpoints[member][nCheckpoints - 1].fromBlock == block.number
+        ) {
+            checkpoints[member][nCheckpoints - 1].amount = newAmount;
+        } else {
+            checkpoints[member][nCheckpoints] = Checkpoint(
+                uint96(block.number),
+                newAmount
+            );
+            numCheckpoints[member] = nCheckpoints + 1;
+        }
+        // slither-disable-next-line reentrancy-events
+        // emit NewBalance(member, token, newAmount);
```

Comment:
I've commented this out for now, is this event necessary here?

Reply:
Actually I think you're right we don't need it